jib-core/src/main/java/com/google/cloud/tools/jib/image/ReproducibleLayerBuilder.java

```diff
@@ -111,12 +111,16 @@ public Blob build() {
       TarArchiveEntry entry =
           new TarArchiveEntry(
               layerEntry.getSourceFile().toFile(), layerEntry.getAbsoluteExtractionPathString());
-      Set<PosixFilePermission> permissions = layerEntry.getPermissions();
-      if (permissions == null) {
-        permissions =
-            Files.isDirectory(layerEntry.getSourceFile())
-                ? defaultFolderPermissions
-                : defaultFilePermissions;
+
+      Set<PosixFilePermission> permissions;
+      if (layerEntry.getPermissions().isPresent()) {
+        permissions = layerEntry.getPermissions().get();
+      } else {
+        if (Files.isDirectory(layerEntry.getSourceFile())) {
+          permissions = defaultFolderPermissions;
+        } else {
+          permissions = defaultFilePermissions;
+        }
```

Comment:
Just saying `permissions = layerEntry.getPermissions().orElse(Files.isDirectory(...) ? defaultFolderPermissions : defaultFilePermissions)` can also work (I think), but not that it's better.